// zero/model/FlipFilter.js
var extend = require('extend');
var config = require('../config');

var FlipFilter = function(filter) {
	if (filter) {
		var f = new FlipFilter();
		filter.pageNo && (f.pageNo = parseInt(filter.pageNo));
		filter.pageSize && (f.pageSize = parseInt(filter.pageSize));
		filter.totalCount && (f.totalCount = parseInt(filter.totalCount));

		for (var key in filter) {
			if(key != '__proto__'){
				f[key] = filter[key]
			}
		}
		return f;
	}
};

FlipFilter.prototype = {
	pageNo: 1,
	pageSize: config.pageSize,
	totalCount: 0,
	totalPage: 1,
	pageStart: 1,
	pageEnd: 1,
	init: function() {
		this.pageStart = this.pageNo - 3 >= 1 ? this.pageNo - 3 : 1;
		this.totalPage = Math.floor(this.totalCount / this.pageSize);
		if (this.totalCount % this.pageSize > 0) {
			this.totalPage++;
		}
		this.totalPage = this.totalPage == 0 ? 1 : this.totalPage;
		this.pageEnd = this.pageNo + 3 <= this.totalPage ? this.pageNo + 3 : this.totalPage;
		if (this.pageNo > this.pageEnd) {
			this.pageNo = this.pageEnd;
		}
		return this;
	},
};

module.exports = FlipFilter;
